fdbserver/workloads/WriteTagThrottling.actor.cpp

```diff
@@ -0,0 +1,262 @@
+	std::unique_ptr<ContinuousSample<double>> badActorCommitLatency, goodActorCommitLatency;
+
+	// Test configuration
+	// KVWorkload::actorCount
+	int goodActorPerClient, badActorPerClient;
+	int numWritePerTx, numReadPerTx, numClearPerTx;
+	int keyCount, hotKeyCount;
+	int sampleSize;
+	double badOpRate;
+	double testDuration;
+	bool writeThrottle;
+
+	// internal states
+	double txBackoffDelay;
+	TransactionTag badReadTag, badWriteTag, goodTag;
+	static constexpr const char* NAME = "WriteTagThrottling";
+
+	WriteTagThrottlingWorkload(WorkloadContext const& wcx) : KVWorkload(wcx) {
+		testDuration = getOption(options, LiteralStringRef("testDuration"), 120.0);
+		badOpRate = getOption(options, LiteralStringRef("badOpRate"), 0.9);
+		numWritePerTx = getOption(options, LiteralStringRef("numWritePerTx"), 1);
+		numReadPerTx = getOption(options, LiteralStringRef("numReadPerTx"), 1);
+		numClearPerTx = getOption(options, LiteralStringRef("numClearPerTx"), 1);
+		sampleSize = getOption(options, LiteralStringRef("sampleSize"), 10000);
+		badActorReadLatency.reset(new ContinuousSample<double>(sampleSize));
+		goodActorReadLatency.reset(new ContinuousSample<double>(sampleSize));
+		badActorCommitLatency.reset(new ContinuousSample<double>(sampleSize));
+		goodActorCommitLatency.reset(new ContinuousSample<double>(sampleSize));
+
+		writeThrottle = getOption(options, LiteralStringRef("writeThrottle"), false);
+		badActorPerClient = getOption(options, LiteralStringRef("badActorPerClient"), 1);
+		goodActorPerClient = getOption(options, LiteralStringRef("goodActorPerClient"), 1);
+		actorCount = goodActorPerClient + badActorPerClient;
+
+		keyCount = getOption(options, LiteralStringRef("keyCount"), 3000);
+		txBackoffDelay = actorCount * 1.0 / getOption(options, LiteralStringRef("txPerSecond"), 1000);
+
+		badReadTag = TransactionTag(std::string("badReadTag"));
+		badWriteTag = TransactionTag(std::string("badWriteTag"));
+		goodTag = TransactionTag(std::string("goodTag"));
+	}
+
+	virtual std::string description() { return WriteTagThrottlingWorkload::NAME; }
+
+	// choose a tag
+	ACTOR static Future<Void> _setup(Database cx, WriteTagThrottlingWorkload* self) {
+		state TransactionPriority priority = deterministicRandom()->randomChoice(allTransactionPriorities);
+		state double rate = deterministicRandom()->random01() * 20;
+		state TagSet tagSet1;
+		state TagSet tagSet2;
+		state TagSet tagSet3;
+		tagSet1.addTag(self->badWriteTag);
+		tagSet2.addTag(self->badReadTag);
+		tagSet3.addTag(self->goodTag);
+
+		wait(ThrottleApi::throttleTags(cx, tagSet1, rate, self->testDuration + 120, TagThrottleType::MANUAL, priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet2, rate, self->testDuration + 120, TagThrottleType::MANUAL, priority));
+		wait(ThrottleApi::throttleTags(cx, tagSet3, rate, self->testDuration + 120, TagThrottleType::MANUAL, priority));
+
+		return Void();
+	}
+	Future<Void> setup(const Database& cx) override { return _setup(cx, this); }
```

Comment:
```suggestion
	Future<Void> setup(const Database& cx) override { return clientId ? Void() : _setup(cx, this); }
```

Reply:
Only one client should run the `_setup` function here, so the convention is to only use the client with `clientId == 0`.